### PR TITLE
refactor bucket storage file open

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
working on speeding up startup.
Prepare to load disk index files from previous run instead of always creating them from scratch.

#### Summary of Changes
Refactor

Fixes #

### CODE REVIEW

File: bucket_map/src/bucket_storage.rs [hunk 1]
```diff
@@ -337,57 +337,79 @@ impl<O: BucketOccupied> BucketStorage<O> {
...
+            data.write_all(&[0]).unwrap();
+            data.rewind().unwrap();
+            measure_new_file.stop();
+            let measure_flush = Measure::start("measure_flush");
+            data.flush().unwrap(); // can we skip this?
+            stats
+                .flush_file_us
+                .fetch_add(measure_flush.end_as_us(), Ordering::Relaxed);
+        }
         let mut measure_mmap = Measure::start("measure_mmap");
```
Comment (HaoranYi) [inline bucket_map/src/bucket_storage.rs:383]: nit: maybe assert when !create, file.size >= create_bytes.

Comment (jeffwashington) [reply bucket_map/src/bucket_storage.rs:383]: `create_bytes` only makes sense if we are creating the file. We do now panic if we can't mmap. before we just failed on `unwrap`.

Comment (HaoranYi) [reply bucket_map/src/bucket_storage.rs:383]: ok.

**Review by HaoranYi — APPROVED** [review]
lgtm

File: bucket_map/src/bucket_storage.rs [hunk 2]
```diff
@@ -337,57 +337,79 @@ impl<O: BucketOccupied> BucketStorage<O> {
...
             .fetch_add(measure_mmap.as_us(), Ordering::Relaxed);
-        res
+        res.ok()
+    }
+
+    /// allocate a new memory mapped file of size `bytes` on one of `drives`
+    fn new_map(drives: &[PathBuf], bytes: u64, stats: &BucketStats) -> (MmapMut, PathBuf, u128) {
+        let r = thread_rng().gen_range(0..drives.len());
+        let drive = &drives[r];
+        let file_random = thread_rng().gen_range(0..u128::MAX);
```
Comment (HaoranYi) [inline bucket_map/src/bucket_storage.rs:407]: If `file_random` produces the same number in two calls, `map_open_file` will panic.
This is very unlikely. Maybe we don't need to handle this case. Anyway just a note.

**Pull request merged**